Classes/EventStore/Storage/Doctrine/DoctrineEventStorage.php

```diff
@@ -79,125 +74,72 @@ public function load(string $streamName)
             ->addOrderBy('version', 'ASC')
             ->setParameter('stream', $streamName, \PDO::PARAM_STR);
 
-        $data = $this->unserializeEvents($query);
-
-        if ($data === []) {
-            return null;
-        }
-
-        $cacheKey = md5($streamName . '.' . $version);
-        $this->runtimeCache[$cacheKey] = new EventStreamData($data, $version);
-
-        return $this->runtimeCache[$cacheKey];
+        $streamIterator = new DoctrineStreamIterator($query->execute());
+        return new EventStream($streamIterator, $this->getStreamVersion($streamName));
```

Comment:
Previously *all* events where loaded from the stream and unserialized here. This won't work for a large amount of events.
Now this returns a custom iterator that will fetch the data lazily and can be optimized internally (i.e. using pagination to fetch x records at a time)